### PR TITLE
Fix issues with components reuse

### PR DESCRIPTION
**Pull request by mvlasieva**
This PR fixes a couple of issues with reusing both parent and child components. In makes sure components and collection view cells can be reused independently and cells in the collection view cache which are still holding a reference to already reused components don't interfere with them.
@JohnSundell

### CODE REVIEW

File: sources/HUBViewController.m [hunk 1]
```diff
@@ -708,19 +706,16 @@ - (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cell
...
-        [self didAddComponentWrapper:componentWrapper];
-    }
-    
-    HUBComponentWrapper * const componentWrapper = [self componentWrapperFromCell:cell];
+
+    HUBComponentWrapper * const componentWrapper = [self.componentReusePool componentWrapperForModel:componentModel
+                                                                                            delegate:self
+                                                                                              parent:nil];
+
+    self.componentWrappersByCellIdentifier[cell.identifier] = componentWrapper;
```
Comment (JohnSundell) [inline sources/HUBViewController.m]: Are we ever clearing this up? That is, before we assign a new wrapper to the cell, we should do:
```
self.componentWrappersByCellIdentifier[cell.identifier] = nil;
```

Comment (mvlasieva) [inline sources/HUBViewController.m]: Sorry, I don't get it. What is the point of removing object from the dictionary right before setting a new object for the same key?

Comment (JohnSundell) [inline sources/HUBViewController.m]: You are completely right, sorry - I was confused 😆

**Comment by JohnSundell** [comment]
@mvlasieva Since Travis are currently undergoing an incident, I'm using my computer as our build server 😅 However, this PR doesn't pass the tests locally. Can you update the PR so that they do? 🙂

**Comment by mvlasieva** [comment]
@JohnSundell All done, feel free to review.

**Pull request merged**